### ISSUE TITLE
Consider python 3.5 async statement

### ISSUE THREAD

**Issue by nramirezuy** [bps/vim-textobj-python#20]
```python
async def function():
    pass


class Foo(object):

    async def method():
        pass
```
source: [PEP-0492](https://www.python.org/dev/peps/pep-0492/#new-coroutine-declaration-syntax)

WORKAROUND:
```diff
diff --git a/autoload/textobj/python.vim b/autoload/textobj/python.vim
index 4f5b146..48b3658 100644
--- a/autoload/textobj/python.vim
+++ b/autoload/textobj/python.vim
@@ -177,10 +177,18 @@ function! textobj#python#class_select_i()
 endfunction
 
 function! textobj#python#function_select_a()
+    let l:async = textobj#python#select_a('async')
+    if type(l:async) == type([])
+        return l:async
+    endif
     return textobj#python#select_a('def')
 endfunction
 
 function! textobj#python#function_select_i()
+    let l:async = textobj#python#select_i('async')
+    if type(l:async) == type([])
+        return l:async
+    endif
     return textobj#python#select_i('def')
 endfunction
```
Even tho I found this workaround I still not confident in it to be the final fix.

**Comment by nramirezuy** [comment]
Just found that this works a lot better :smile: 

```diff
diff --git a/after/ftplugin/python/textobj-python.vim b/after/ftplugin/python/textobj-python.vim
index 5a86922..47ee559 100644
--- a/after/ftplugin/python/textobj-python.vim
+++ b/after/ftplugin/python/textobj-python.vim
@@ -56,7 +56,7 @@ call textobj#user#plugin('python', {
 \       'select-i': '<buffer>if',
 \       'select-a-function': 'textobj#python#function_select_a',
 \       'select-i-function': 'textobj#python#function_select_i',
-\       'pattern': '^\s*\zsdef \(.\|\n\)\{-}:',
+\       'pattern': '^\s*\zs\(def\|async def\) \(.\|\n\)\{-}:',
 \       'move-p': '<buffer>[pf',
 \       'move-n': '<buffer>]pf',
 \   }
diff --git a/autoload/textobj/python.vim b/autoload/textobj/python.vim
index 4f5b146..555b8ee 100644
--- a/autoload/textobj/python.vim
+++ b/autoload/textobj/python.vim
@@ -177,11 +177,11 @@ function! textobj#python#class_select_i()
 endfunction
 
 function! textobj#python#function_select_a()
-    return textobj#python#select_a('def')
+    return textobj#python#select_i('\(async def\|def\)')
 endfunction
 
 function! textobj#python#function_select_i()
-    return textobj#python#select_i('def')
+    return textobj#python#select_i('\(async def\|def\)')
 endfunction
 
 function! textobj#python#function_select(object_type)
```

**Comment by bps** [comment]
Thank you for the patch! Please feel free to submit a PR with unit tests. If you'd rather not do that, I will get to this sometime in the coming weeks.

**Comment by bps** [comment]
Resolved in #21. Thanks for the initial request!
